tests/shell_tests.bzl

```diff
@@ -83,19 +83,17 @@ def _shell_spawn_e2e_test_impl(ctx):
         "line tab\tcharacter $foo qu\"o\"te it'\\''s foo\\bar back`echo q`uote'",
         '[[ "${output}" == "${expected}" ]]',
     ])
-    script_file = ctx.actions.declare_file("%s.sh" % (ctx.label.name))
+    script_file = ctx.outputs.script
     ctx.actions.write(
         output = script_file,
         content = script_content,
         is_executable = True,
     )
-    return [
-        DefaultInfo(executable = script_file),
-    ]
+    return [DefaultInfo(files = depset([script_file]))]
 
-shell_spawn_e2e_test = rule(
-    test = True,
-    implementation = _shell_spawn_e2e_test_impl,
+shell_args_test_gen = rule(
+    implementation = _shell_args_test_gen_impl,
+    outputs = {"script": "%{name}.sh"},
```

Comment:
This syntax is subject to deprecation soon (TM)
See https://docs.google.com/document/d/14A9HK8Jn2jErMayLEE3nrNJIxNfZWN_slFbhgtS6-aM 

Could you use the recommended alternatives? (You can use `attr.output` and set the value with a macro, or just return a file as part of the returned DefaultInfo if you don't need to reference the output as its own label)

Reply:
Done.